src/movies.js

```diff
@@ -1,15 +1,107 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(array){
+    let directors = array.map((movies) => {
+    return movies.director
+    })
+ return directors
+}
+getAllDirectors(movies)
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(array) {
+    let dramaMoviesSpielberg = array.filter((movies) =>{
+         return (movies.director === "Steven Spielberg") && (movies.genre.includes("Drama"))       
+    })
+    return dramaMoviesSpielberg.length
+}
+
+howManyMovies(movies)
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(array) {
+    let allRates = array.reduce((acc, elem) => {
+        return acc + elem.rate
+      },0)
+                                
+      let avgRates = allRates/array.length
+      return avgRates.toFixed(2)
```

Comment:
the .toFixed() is kinda of an odd method because it takes a number to reduce the amount of decimals, however it will return the number as a string. So you will need to manually change the string back to a number, you can do this in different ways, below there are 3 common ways.
````
Number(avgRates)

parseFloat(avgRates)
+avgRates
````